cognite/powerops/client/shop_api.py

```diff
@@ -90,34 +135,92 @@ def update(self):
 
     def list(self) -> list[ShopModel]:
-        return ShopModel()
+        return [ShopModel()]
 
     def retrieve(self, model_id):
         m = ShopModel()
         m.model_id = model_id
         return m
 
 
-class ShopRunsApi:
-    def __init__(self, shop_api: "ShopAPI"):
-        self._shop_api = shop_api
+class ShopRunsAPI:
+    def __init__(self, po_client: PowerOpsClient):
+        self._po_client = po_client
 
-    def trigger(self, case) -> ShopRun:
-        return ShopRun(case=case)
+    def trigger(self, watercourse_name: str, case: Case) -> ShopRun:
+        logger.info(f"Triggering SHOP run with watercourse name {watercourse_name}")
+        shop_run_event = ShopRunEvent(
+            watercourse=watercourse_name,
+            starttime=case["time.starttime"],
+            endtime=case["time.endtime"],
+            timeresolution=case["time.timeresolution"],
+        )
+        shop_run = self._upload_to_cdf(watercourse_name, shop_run_event, case)
+        self._post_shop_run(shop_run_event)
+        return shop_run
+
+    def _upload_to_cdf(self, watercourse_name: str, shop_run_event: ShopRunEvent, case: Case) -> ShopRun:
+        file_ext_id = f"{shop_run_event.external_id}_CASE_FILE"
+        logger.debug(f"Uploading case file '{file_ext_id}'.")
+        file_meta = self._po_client.cdf.files.upload_bytes(
+            external_id=file_ext_id,
+            content=case.yaml.encode(),
+            data_set_id=self._po_client.write_dataset_id,
+            name=f"{watercourse_name}_case.yaml",
+            mime_type="application/yaml",
+            metadata={
+                "shop:run_event_id": shop_run_event.external_id,
+                "shop:type": "cog_shop_case",
+            },
+            overwrite=True,
+        )
+        logger.debug(f"Uploading event '{shop_run_event.external_id}'.")
+        event = shop_run_event.to_event(self._po_client.write_dataset_id)
+        event.metadata["shop:preprocessor_data"] = json.dumps({"cog_shop_case_file": file_ext_id})
+        self._po_client.cdf.events.create(event)
+        self._po_client.cdf.relationships.create(
+            simple_relationship(
+                source=event,
+                target=file_meta,
+                label_external_id=RelationshipLabels.CASE_FILE,
+            )
+        )
+        return ShopRun(self._po_client, shop_run_event=shop_run_event)
+
+    def _post_shop_run(self, shop_run_event: ShopRunEvent):
+        logger.debug("Triggering run-shop endpoint.")
+        cdf_config = self._po_client.cdf.config
+        project = cdf_config.project
+        cluster = cdf_config.base_url.split("//")[1].split(".")[0]
+        url = f"https://power-ops-api.staging.{cluster}.cognite.ai/{project}" f"/run-shop"
+        auth_header = dict([cdf_config.credentials.authorization_header()])
+
+        response = requests.post(
+            url,
+            json={"shopEventExternalId": shop_run_event.external_id, "cogShopVersion": "TEST123"},  # image version
```

Comment:
Should the `TEST123` be hardcoded here? Expect this to come from something like a global config.